Realm/ObjectStore/handover.cpp

```diff
@@ -0,0 +1,190 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+// This is lets our internals access AnyHandover's private constructors.
+#define private public
+#include "handover.hpp"
+#undef private
+
+using namespace realm;
+
+#pragma mark AnyThreadConfined
+
+AnyThreadConfined::AnyThreadConfined(const AnyThreadConfined& thread_confined) {
+    switch (thread_confined.m_type) {
+        case Type::Row:
+            new (&m_row) Row(thread_confined.m_row);
+            break;
+
+        case Type::Query:
+            new (&m_query) Query(thread_confined.m_query);
+            break;
+
+        case Type::TableRef:
+            new (&m_table_ref) TableRef(thread_confined.m_table_ref);
+            break;
+
+        case Type::TableView:
+            new (&m_table_view) TableView(thread_confined.m_table_view);
+            break;
+
+        case Type::LinkViewRef:
+            new (&m_link_view_ref) LinkViewRef(thread_confined.m_link_view_ref);
+            break;
+    }
+    new (&m_type) Type(thread_confined.m_type);
+}
+
+AnyThreadConfined::AnyThreadConfined(AnyThreadConfined&& thread_confined) {
+    switch (thread_confined.m_type) {
+        case Type::Row:
+            new (&m_row) Row(std::move(thread_confined.m_row));
+            break;
+
+        case Type::Query:
+            new (&m_query) Query(std::move(thread_confined.m_query));
+            break;
+
+        case Type::TableRef:
+            new (&m_table_ref) TableRef(std::move(thread_confined.m_table_ref));
+            break;
+
+        case Type::TableView:
+            new (&m_table_view) TableView(std::move(thread_confined.m_table_view));
+            break;
+
+        case Type::LinkViewRef:
+            new (&m_link_view_ref) LinkViewRef(std::move(thread_confined.m_link_view_ref));
+            break;
+    }
+    new (&m_type) Type(std::move(thread_confined.m_type));
+}
+
+AnyThreadConfined::~AnyThreadConfined() {
+    switch (m_type) {
+        case Type::Row:
+            m_row.~Row();
+            break;
+
+        case Type::Query:
+            m_query.~Query();
+            break;
+
+        case Type::TableRef:
+            m_table_ref.~TableRef();
+            break;
+
+        case Type::TableView:
+            m_table_view.~TableView();
+            break;
+
+        case Type::LinkViewRef:
+            m_link_view_ref.~LinkViewRef();
+            break;
+    }
+}
+
+AnyHandover AnyThreadConfined::export_for_handover(SharedGroup &shared_group) const {
+    switch (m_type) {
+        case AnyThreadConfined::Type::Row:
+            return AnyHandover (shared_group.export_for_handover(m_row));
```

Comment:
Only language control-flow things have a space between the name and opening parenthesis.